### ISSUE TITLE
New added node remained in `UJ` status on the part of nodes. `Host ID` is NULL

### ISSUE THREAD

**Issue by juliayakovlev** [scylladb/scylladb#16902]
## Issue description

- [ ] This issue is a regression.
- [x] It is unknown if this issue is a regression.

New added node remained in **`UJ` status** on the part of the nodes. **`Host ID` is NULL**
```
< t:2024-01-20 14:31:26,423 f:cluster.py      Node longevity-tls-50gb-3d-master-db-node-a47195bb-3 [3.249.197.113 | 10.4.11.78] (seed: True): Command '/usr/bin/nodetool -u cassandra -pw 'cassandra'  status ' duration -> 4.609257415999309 s
< t:2024-01-20 14:31:25,881  > Datacenter: eu-west
< t:2024-01-20 14:31:25,881  > ===================
< t:2024-01-20 14:31:25,881  > Status=Up/Down
< t:2024-01-20 14:31:25,881  > |/ State=Normal/Leaving/Joining/Moving
< t:2024-01-20 14:31:25,892  > --  Address      Load       Tokens       Owns    Host ID                               Rack
< t:2024-01-20 14:31:25,892  > UN  10.4.11.148  16.78 GB   256          ?       eb886165-554f-4f83-a9c6-bc96c50703aa  1c
< t:2024-01-20 14:31:25,898  > UN  10.4.9.53    23.37 GB   256          ?       a7276127-9837-496a-af5d-d51f4b8a995f  1c
< t:2024-01-20 14:31:25,898  > UN  10.4.11.128  20.51 GB   256          ?       eb4da176-aa33-4808-a234-1273ecbcde68  1c
< t:2024-01-20 14:31:25,903  > UJ  10.4.10.195  8.2 GB     256          ?       null                                  1c
< t:2024-01-20 14:31:25,904  > UN  10.4.11.204  11.41 GB   256          ?       c3f63004-c652-42d7-bb82-f32d5d2f9206  1c
< t:2024-01-20 14:31:25,919  > UN  10.4.8.254   14.81 GB   256          ?       d362b602-62c7-4023-91e9-88959c60b344  1c
< t:2024-01-20 14:31:25,919  > UN  10.4.11.78   20.82 GB   256          ?       387001f8-8d54-4ecd-85b9-f52ef36dc506  1c
< t:2024-01-20 14:31:25,919  > UN  10.4.11.186  14.89 GB   256          ?       b2ed0345-894e-4a99-9c81-6bcd8cc849f8  1c
< t:2024-01-20 14:31:25,919  > UN  10.4.8.91    21.99 GB   256          ?       48459c32-5a4a-4a00-9b94-5146461cb8f0  1c
```

The reason is not clear

New node `longevity-tls-50gb-3d-master-db-node-a47195bb-13` was added successfully. 
Bootstrap completed and Scylla was up:
```
Jan 20 14:29:52.978898 longevity-tls-50gb-3d-master-db-node-a47195bb-13 scylla[6071]:  [shard  0:strm] storage_service - Bootstrap completed! for the tokens {-893713186037081828, 95028400449033684, -825592331035
Jan 20 14:29:53.270496 longevity-tls-50gb-3d-master-db-node-a47195bb-13 scylla[6071]:  [shard  0:strm] storage_service - Node 10.4.10.195 is in normal state, tokens: {7272125785311453289, -7307171860973556829, -7872304161031330464, -2363689394615674191, -5764778108122059983, -7840521465712546653, 5806842588930604289, -2088427809536837340, 4103143574413007024, 1002782137723845030, -2565294486575191857, 30404839657599046

Jan 20 14:30:05.490245 longevity-tls-50gb-3d-master-db-node-a47195bb-13 scylla[6071]:  [shard  0:main] init - Scylla version 5.5.0~dev-0.20240119.b1ba904c4977 initialization completed.
```

4 nodes (including the new node himself) see this node state as normal.
**10.4.10.195**  - it is private IP of the new added node.

I found in the logs of those 4 nodes the message:
```
longevity-tls-50gb-3d-master-db-node-a47195bb-11 scylla[6075]:  [shard  0: gms] 
storage_service - Node 10.4.10.195 is in normal state, tokens: {7272125785311453289, -7307171860973556829, -7872304161031330464, 
```
```
longevity-tls-50gb-3d-master-db-node-a47195bb-2 scylla[9247]:  [shard  0: gms] 
storage_service - Node 10.4.10.195 is in normal state, tokens: {7272125785311453289, -7307171860973556829, -7872304161031330464, -2363689394615674191
```

```
longevity-tls-50gb-3d-master-db-node-a47195bb-9 scylla[758]:  [shard  0: gms] 
storage_service - Node 10.4.10.195 is in normal state, tokens: {7272125785311453289, -7307171860973556829, -7872304161031330464,
```

```
longevity-tls-50gb-3d-master-db-node-a47195bb-13 scylla[6071]:  [shard  0:strm] 
storage_service - Node 10.4.10.195 is in normal state, tokens: {7272125785311453289, -7307171860973556829, -7872304161031330464, -2363689394615674191,
```

In logs of **other nodes last message about 10.4.10.195** is:
```
Jan 20 14:27:02.271531 longevity-tls-50gb-3d-master-db-node-a47195bb-3 scylla[760]:  [shard  0: gms] gossip - InetAddress 10.4.10.195 is now UP, status = UNKNOWN
Jan 20 14:27:02.609885 longevity-tls-50gb-3d-master-db-node-a47195bb-3 scylla[760]:  [shard  0:main] raft_group_registry - marking Raft server 741fdc78-2a04-4cbf-831e-cf7827a1daa5 as alive for raft groups
Jan 20 14:27:52.336049 longevity-tls-50gb-3d-master-db-node-a47195bb-3 scylla[760]:  [shard  0:strm] storage_service - bootstrap[e69e83c9-5e46-47e1-a2a0-d1b836d76c7a]: Added node=10.4.10.195/741fdc78-2a04-4cbf-831e-cf7827a1daa5 as bootstrap, coordinator=10.4.10.195/741fdc78-2a04-4cbf-831e-cf7827a1daa5
Jan 20 14:27:53.405141 longevity-tls-50gb-3d-master-db-node-a47195bb-3 scylla[760]:  [shard  0: gms] cdc - Starting to use generation (2024/01/20 14:28:57, 3fb3200f-e058-43dd-b5e0-4fc3f01ca205)
Jan 20 14:27:53.405827 longevity-tls-50gb-3d-master-db-node-a47195bb-3 scylla[760]:  [shard  0: gms] cdc - Generation (2024/01/20 14:28:57, 3fb3200f-e058-43dd-b5e0-4fc3f01ca205): streams description table already updated.
Jan 20 14:29:52.973568 longevity-tls-50gb-3d-master-db-node-a47195bb-3 scylla[760]:  [shard  0:strm] 
storage_service - bootstrap[e69e83c9-5e46-47e1-a2a0-d1b836d76c7a]: Marked ops done from coordinator=10.4.10.195
```

The status remain **UNKNOWN** and there is no other information during 20 minutes, up to test finish.

I did not find errors in the logs.

There `C++ Exceptions` on all instances (except of new one) in this time. I am sure that it is important
![Screenshot from 2024-01-21 18-22-28](https://github.com/scylladb/scylladb/assets/34435448/29594dc6-6d11-469f-8b3d-ccfb1326b59c)

Right before adding the node `disrupt_mgmt_repair_cli` nemesis was triggered. This nemesis performed successfully `sudo sctool repair -c a5f9cc9b-b111-4d26-bd58-86eff85af55e` command.

## Impact

Cluster is not ready

## How frequently does it reproduce?

Need to try to reproduce it

## Installation details

Kernel Version: 5.15.0-1051-aws
Scylla version (or git commit hash): `5.5.0~dev-20240119.b1ba904c4977` with build-id `7a5829efb1f6ef7b467d2dc837300abcc0b739c8`

Cluster size: 6 nodes (i4i.4xlarge)

Scylla Nodes used in this run:
- longevity-tls-50gb-3d-master-db-node-a47195bb-9 (3.254.177.96 | 10.4.11.186) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-8 (34.245.83.217 | 10.4.8.76) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-7 (54.220.160.172 | 10.4.8.91) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-6 (54.170.8.126 | 10.4.11.128) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-5 (3.252.245.134 | 10.4.8.120) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-4 (34.241.41.74 | 10.4.9.53) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-3 (3.249.197.113 | 10.4.11.78) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-2 (34.246.160.157 | 10.4.11.148) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-13 (34.244.5.23 | 10.4.10.195) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-12 (54.170.193.150 | 10.4.11.204) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-11 (18.201.204.111 | 10.4.8.254) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-10 (18.201.217.35 | 10.4.10.105) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-a47195bb-1 (18.201.192.154 | 10.4.9.91) (shards: 14)

OS / Image: `ami-0c85f335032e27007` (aws: undefined_region)

Test: `longevity-50gb-3days-test`
Test id: `a47195bb-a06b-415a-89a3-b4fe8a424fcd`
Test name: `scylla-master/longevity/longevity-50gb-3days-test`
Test config file(s):

- [longevity-50GB-3days-authorization-and-tls-ssl.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/d311c6c14f705175817159c0bce264a17e644447/test-cases/longevity/longevity-50GB-3days-authorization-and-tls-ssl.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor a47195bb-a06b-415a-89a3-b4fe8a424fcd`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=a47195bb-a06b-415a-89a3-b4fe8a424fcd)
- Show all stored logs command: `$ hydra investigate show-logs a47195bb-a06b-415a-89a3-b4fe8a424fcd`

## Logs:
- **db-cluster-a47195bb.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/db-cluster-a47195bb.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/db-cluster-a47195bb.tar.gz)
- **sct-runner-events-a47195bb.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/sct-runner-events-a47195bb.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/sct-runner-events-a47195bb.tar.gz)
- **sct-a47195bb.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/sct-a47195bb.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/sct-a47195bb.log.tar.gz)
- **loader-set-a47195bb.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/loader-set-a47195bb.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/loader-set-a47195bb.tar.gz)
- **monitor-set-a47195bb.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/monitor-set-a47195bb.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a47195bb-a06b-415a-89a3-b4fe8a424fcd/20240120_150300/monitor-set-a47195bb.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-50gb-3days-test/232/)
[Argus](https://argus.scylladb.com/test/5df88bc6-3ede-4a28-9586-773f93374239/runs?additionalRuns[]=a47195bb-a06b-415a-89a3-b4fe8a424fcd)
</details>

**Comment by aleksbykov** [comment]
The same happened during another job:
## Issue description

after node5 was decommisisoned and node8 was bootsrapped. node1 see this node8 state as UJ with null host id.
At same time there a lot of reactor stals, the largest reached up to 800 ms:
```
2024-01-27T12:35:09.233+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-2     !INFO | scylla[26159]: Reactor stalled for 845 ms on shard 9. Backtrace: 0x5c934ca 0x5c928d5 0x5c93c8f 0x3dbaf 0x1d83764 0x1f88192 0x1f82321 0x1f67495 0x1e0a3ff 0x3b7fb77 0x3b8109d 0x3b89784 0x3b87733 0x3c00fab 0x3bc00aa 0x5ca50ef 0x5ca63d7 0x5cca150 0x5c6596a 0x8c946 0x11285f

[Backtrace #0]
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace_oneline()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace_oneline()::{lambda(seastar::frame)#1}&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:68
 (inlined by) seastar::backtrace_buffer::append_backtrace_oneline() at ./build/release/seastar/./seastar/src/core/reactor.cc:840
 (inlined by) seastar::print_with_backtrace(seastar::backtrace_buffer&, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:859
seastar::internal::cpu_stall_detector::generate_trace() at ./build/release/seastar/./seastar/src/core/reactor.cc:1473
seastar::internal::cpu_stall_detector::maybe_report() at ./build/release/seastar/./seastar/src/core/reactor.cc:1210
 (inlined by) seastar::internal::cpu_stall_detector::on_signal() at ./build/release/seastar/./seastar/src/core/reactor.cc:1230
 (inlined by) seastar::reactor::block_notifier(int) at ./build/release/seastar/./seastar/src/core/reactor.cc:1511
?? ??:0
compare_atomic_cell_for_merge(atomic_cell_view, atomic_cell_view) at atomic_cell.cc:?
collection_mutation_view_description difference<abstract_type const&>(collection_mutation_view_description, collection_mutation_view_description, abstract_type const&) at ./collection_mutation.cc:336
operator() at ./collection_mutation.cc:355
 (inlined by) std::invoke_result<seastar::internal::variant_visitor<difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0::operator()(collection_mutation_view_description) const::{lambda(collection_mutation_view_description)#1}::operator()(collection_mutation_view_description) const::{lambda(collection_type_impl const&)#1}, difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0::operator()(collection_mutation_view_description) const::{lambda(collection_mutation_view_description)#1}::operator()(collection_mutation_view_description) const::{lambda(user_type_impl const&)#1}, difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0::operator()(collection_mutation_view_description) const::{lambda(collection_mutation_view_description)#1}::operator()(collection_mutation_view_description) const::{lambda(abstract_type const&)#1}>, ascii_type_impl const&>::type visit<seastar::internal::variant_visitor<difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0::operator()(collection_mutation_view_description) const::{lambda(collection_mutation_view_description)#1}::operator()(collection_mutation_view_description) const::{lambda(collection_type_impl const&)#1}, difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0::operator()(collection_mutation_view_description) const::{lambda(collection_mutation_view_description)#1}::operator()(collection_mutation_view_description) const::{lambda(user_type_impl const&)#1}, difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0::operator()(collection_mutation_view_description) const::{lambda(collection_mutation_view_description)#1}::operator()(collection_mutation_view_description) const::{lambda(abstract_type const&)#1}> >(abstract_type const&, {lambda(abstract_type const&)#1}&&) at ././concrete_types.hh:207
operator() at ./collection_mutation.cc:353
 (inlined by) decltype(auto) collection_mutation_view::with_deserialized<difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0::operator()(collection_mutation_view_description) const::{lambda(collection_mutation_view_description)#1}>(abstract_type const&, difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0::operator()(collection_mutation_view_description) const::{lambda(collection_mutation_view_description)#1}) const at ././collection_mutation.hh:95
 (inlined by) operator() at ./collection_mutation.cc:352
 (inlined by) decltype(auto) collection_mutation_view::with_deserialized<difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0>(abstract_type const&, difference(abstract_type const&, collection_mutation_view, collection_mutation_view)::$_0) const at ././collection_mutation.hh:95
 (inlined by) difference(abstract_type const&, collection_mutation_view, collection_mutation_view) at ./collection_mutation.cc:351
row::difference(schema const&, column_kind, row const&) const at ./mutation/mutation_partition.cc:1905
db::view::view_updates::update_entry(data_dictionary::database, partition_key const&, db::view::clustering_or_static_row const&, db::view::clustering_or_static_row const&, std::chrono::time_point<gc_clock, std::chrono::duration<long, std::ratio<1l, 1l> > >) at ./db/view/view.cc:1047
db::view::view_updates::generate_update(data_dictionary::database, partition_key const&, db::view::clustering_or_static_row const&, std::optional<db::view::clustering_or_static_row> const&, std::chrono::time_point<gc_clock, std::chrono::duration<long, std::ratio<1l, 1l> > >) at view.cc:?
db::view::view_update_builder::generate_update(clustering_row&&, std::optional<clustering_row>&&) at ./db/view/view.cc:1278
db::view::view_update_builder::on_results() at ./db/view/view.cc:1389
db::view::view_update_builder::build_some() at ./db/view/view.cc:1244
std::__n4861::coroutine_handle<seastar::internal::coroutine_traits_base<std::optional<utils::chunked_vector<frozen_mutation_and_schema, 131072ul> > >::promise_type>::resume() const at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/coroutine:240
 (inlined by) seastar::internal::coroutine_traits_base<std::optional<utils::chunked_vector<frozen_mutation_and_schema, 131072ul> > >::promise_type::run_and_dispose() at ././seastar/include/seastar/core/coroutine.hh:83
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at ./build/release/seastar/./seastar/src/core/reactor.cc:2666
 (inlined by) seastar::reactor::run_some_tasks() at ./build/release/seastar/./seastar/src/core/reactor.cc:3129
seastar::reactor::do_run() at ./build/release/seastar/./seastar/src/core/reactor.cc:3305
operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:4537
 (inlined by) void std::__invoke_impl<void, seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_0&>(std::__invoke_other, seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_0&) at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:61
 (inlined by) std::enable_if<is_invocable_r_v<void, seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_0&>, void>::type std::__invoke_r<void, seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_0&>(seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_0&) at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:111
 (inlined by) std::_Function_handler<void (), seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_0>::_M_invoke(std::_Any_data const&) at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:290
std::function<void ()>::operator()() const at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
 (inlined by) seastar::posix_thread::start_routine(void*) at ./build/release/seastar/./seastar/src/core/posix.cc:90
?? ??:0
?? ??:0

```
And on more strage thing in logs, that some node constantly tirggered alive/dead for raft registry:
```
2024-01-27T11:43:15.242+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]: kernel callstack:
2024-01-27T11:43:15.992+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]:  [shard  0:main] raft_group_registry - marking Raft server f41721c4-989c-4327-9550-90e3e83519ec as dead for raft groups
2024-01-27T11:43:16.742+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]:  [shard  0:main] raft_group_registry - marking Raft server f41721c4-989c-4327-9550-90e3e83519ec as alive for raft groups
2024-01-27T11:43:18.742+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]:  [shard  0:main] raft_group_registry - marking Raft server f41721c4-989c-4327-9550-90e3e83519ec as dead for raft groups
2024-01-27T11:43:19.242+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]:  [shard  0:main] raft_group_registry - marking Raft server f41721c4-989c-4327-9550-90e3e83519ec as alive for raft groups
2024-01-27T11:43:19.898+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]:  [shard  0:main] raft_group_registry - marking Raft server f41721c4-989c-4327-9550-90e3e83519ec as dead for raft groups
2024-01-27T11:43:19.898+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]:  [shard  0:main] raft_group_registry - marking Raft server f41721c4-989c-4327-9550-90e3e83519ec as alive for raft groups
2024-01-27T11:43:21.492+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]:  [shard  0:main] raft_group_registry - marking Raft server f41721c4-989c-4327-9550-90e3e83519ec as dead for raft groups
2024-01-27T11:43:21.492+00:00 longevity-mv-si-4d-master-db-node-4ab15b97-1     !INFO | scylla[32946]:  [shard  0:main] raft_group_registry - marking Raft server f41721c4-989c-4327-9550-90e3e83519ec as alive for raft groups
2024
```
where  f41721c4-989c-4327-9550-90e3e83519ec  is host id of node 2



## Impact

*Describe the impact this issue causes to the user.*

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details

Kernel Version: 5.15.0-1052-aws
Scylla version (or git commit hash): `5.5.0~dev-20240125.03313d359e32` with build-id `3ba319c2856389fd41198911105d86848352105e`





Cluster size: 5 nodes (i4i.8xlarge)

Scylla Nodes used in this run:
- longevity-mv-si-4d-master-db-node-4ab15b97-1 (3.249.31.188 | 10.4.8.195) (shards: 30)
- longevity-mv-si-4d-master-db-node-4ab15b97-2 (54.229.39.191 | 10.4.8.215) (shards: 30)
- longevity-mv-si-4d-master-db-node-4ab15b97-3 (54.194.132.182 | 10.4.9.2) (shards: 30)
- longevity-mv-si-4d-master-db-node-4ab15b97-4 (46.137.33.14 | 10.4.11.33) (shards: 30)
- longevity-mv-si-4d-master-db-node-4ab15b97-5 (34.240.225.131 | 10.4.11.228) (shards: 30)
- longevity-mv-si-4d-master-db-node-4ab15b97-6 (3.255.105.104 | 10.4.9.105) (shards: 30)
- longevity-mv-si-4d-master-db-node-4ab15b97-7 (18.201.148.33 | 10.4.10.63) (shards: 30)
- longevity-mv-si-4d-master-db-node-4ab15b97-8 (34.246.172.148 | 10.4.8.205) (shards: 30)


OS / Image: `ami-0ef9a24dac72b0573` (aws: undefined_region)

Test: `longevity-mv-si-4days-test`
Test id: `4ab15b97-5b91-4760-a4bf-c08b0123e637`
Test name: `scylla-master/longevity/longevity-mv-si-4days-test`
Test config file(s):

- [longevity-mv-si-4days.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/cf79ff757eb5b3431035e688ede85c4ea1dae417/test-cases/longevity/longevity-mv-si-4days.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor 4ab15b97-5b91-4760-a4bf-c08b0123e637`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=4ab15b97-5b91-4760-a4bf-c08b0123e637)
- Show all stored logs command: `$ hydra investigate show-logs 4ab15b97-5b91-4760-a4bf-c08b0123e637`


## Logs:
- **db-cluster-4ab15b97.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/db-cluster-4ab15b97.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/db-cluster-4ab15b97.tar.gz)
- **sct-runner-events-4ab15b97.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/sct-runner-events-4ab15b97.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/sct-runner-events-4ab15b97.tar.gz)
- **sct-4ab15b97.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/sct-4ab15b97.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/sct-4ab15b97.log.tar.gz)
- **loader-set-4ab15b97.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/loader-set-4ab15b97.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/loader-set-4ab15b97.tar.gz)
- **monitor-set-4ab15b97.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/monitor-set-4ab15b97.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4ab15b97-5b91-4760-a4bf-c08b0123e637/20240127_132333/monitor-set-4ab15b97.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-mv-si-4days-test/33/)
[Argus](https://argus.scylladb.com/test/b49dc529-c8e7-4117-a01d-2c6669c4dc05/runs?additionalRuns[]=4ab15b97-5b91-4760-a4bf-c08b0123e637)
</details>

**Comment by mykaul** [comment]
@aleksbykov - please open a separate issue on this stall.

**Comment by mykaul** [comment]
@kbr-scylla - could it be related somehow to host ID changes?...

**Comment by kbr-scylla** [comment]
Similar symptoms to https://github.com/scylladb/scylladb/issues/16668. In that issue, a node finishes replace operation but never seemed to transition to NORMAL from the point of view of other nodes. I suspected that some gossiper notifications are not delivered or handled incorrectly. Here we see a bootstrapping node not transitioning to NORMAL. But the missing Host ID in `nodetool status` is a new thing.

I see this build includes 5f44ae8371a382e51eb4bd7aa85628722bd528b6 so maybe I can find some clues in the logs

**Comment by kbr-scylla** [comment]
In fact this "Node 10.4.10.195 is in normal state" which we only see on 4 nodes and for some reason not on others is coming from https://github.com/scylladb/scylladb/commit/5f44ae8371a382e51eb4bd7aa85628722bd528b6 which I sent to investigate https://github.com/scylladb/scylladb/issues/16668. So it is a clue already. Looks like `handle_state_normal` is not called at all on those other nodes.

**Comment by kbr-scylla** [comment]
But apparently my suspicion about gossiper waiting on a lock is also false, we don't see "waiting for endpoint lock" in the logs.

**Comment by kbr-scylla** [comment]
Maybe node 3 gets the gossiper notification but it got broken somehow, with Host ID = null (as evidenced by the nodetool status output) and also STATUS = null

**Comment by kbr-scylla** [comment]
I'm looking for clues in git log. There were some non-trivial changes in how gossiper notifications are delivered such as ad8a9104d859124447d847eeecf9e32c24da9e7c, in December, could be related

**Comment by kbr-scylla** [comment]
The reason Host ID is shown as null for the joining node is because `nodetool status` uses the `/storage_service/host_id` HTTP endpoint, which uses `token_metadata::get_endpoint_to_host_id_map_for_reading()` underneath. If a node is not included in the map returned by this function, `status` will show its host ID has `null`.

This function was changed in https://github.com/scylladb/scylladb/pull/14844 to only return normal token owners. (cc @bhalevy)

So at least this part is a regression introduced by https://github.com/scylladb/scylladb/pull/14844.

To restore the old behavior, we need to change `/storage_service/host_id` endpoint to return all known nodes -- not only normal token owners. If we want `get_endpoint_to_host_id_map_for_reading()` to keep returning only normal nodes (**but** I don't know if it makes sense, especially in raft-topology mode where we have a state where we read from new replicas that might contain a bootstrapping node), we need to use a different function.

The investigation is not over yet, it's still a mistery why `handle_state_normal` wasn't called.

**Comment by kbr-scylla** [comment]
We can infer that the node is present in `token_metadata` (and `locator::topology`) as an owner of bootstrapping token: to show a node as `J`, `nodetool status` uses `/storage_service/nodes/joining` endpoint, which combines `token_metadata::get_bootstrap_tokens()` + with `token_metadata::get_endpoint_for_host_id()`.

**Comment by kbr-scylla** [comment]
`gossipinfo` about our problematic node after it booted:
- on node 2
```
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG > /10.4.10.195
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   generation:1705760821
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   heartbeat:3329
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X8:v2;1705760937261;3fb3200f-e058-43dd-b5e0-4fc3f01ca205
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X1:AGGREGATE_STORAGE_OPTIONS,ALTERNATOR_TTL,CDC,CDC_GENERATIONS_V2,COLLECTION_INDEXING,COMPUTED_COLUMNS,CORRECT_COUNTER_ORDER,CORRECT_IDX_TOKEN_IN_SECONDARY_INDEX,CORRECT_NON_COMPOUND_RANGE_TOMBSTONES,CORRECT_STATIC_COMPACT_IN_MC,COUNTERS,DIGEST_FOR_NULL_VALUES,DIGEST_INSENSITIVE_TO_EXPIRY,DIGEST_MULTIPARTITION_READ,EMPTY_REPLICA_MUTATION_PAGES,EMPTY_REPLICA_PAGES,GROUP0_SCHEMA_VERSIONING,HINTED_HANDOFF_SEPARATE_CONNECTION,INDEXES,LARGE_COLLECTION_DETECTION,LARGE_PARTITIONS,LA_SSTABLE_FORMAT,LWT,MATERIALIZED_VIEWS,MC_SSTABLE_FORMAT,MD_SSTABLE_FORMAT,ME_SSTABLE_FORMAT,NONFROZEN_UDTS,PARALLELIZED_AGGREGATION,PER_TABLE_CACHING,PER_TABLE_PARTITIONERS,RANGE_SCAN_DATA_VARIANT,RANGE_TOMBSTONES,ROLES,ROW_LEVEL_REPAIR,SCHEMA_COMMITLOG,SCHEMA_TABLES_V3,SECONDARY_INDEXES_ON_STATIC_COLUMNS,SEPARATE_PAGE_SIZE_AND_SAFETY_LIMIT,STREAM_WITH_RPC_STREAM,SUPPORTS_RAFT_CLUSTER_MANAGEMENT,TABLE_DIGEST_INSENSITIVE_TO_EXPIRY,TOMBSTONE_GC_OPTIONS,TRUNCATION_TABLE,TYPED_ERRORS_IN_READ_RPC,UDA,UDA_NATIVE_PARALLELIZED_AGGREGATION,UNBOUNDED_RANGE_TOMBSTONES,UUID_SSTABLE_IDENTIFIERS,VIEW_VIRTUAL_COLUMNS,WRITE_FAILURE_REPLY,XXHASH
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X9:org.apache.cassandra.locator.Ec2Snitch
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X3:3
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   LOAD:12929538779
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X5:1062:877238681:1705761990413
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   SCHEMA:a4e0dc54-b797-11ee-f6f5-a538f551ea6e
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   STATUS:NORMAL,-893713186037081828
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X2:system_auth.roles:1.000000;system_traces.node_slow_log:0.000000;system_distributed.cdc_generation_timestamps:0.000000;mview.users:0.992747;system_traces.sessions_time_idx:0.000000;system_auth.role_attributes:0.000000;mview.users_by_last_name:0.574181;system_traces.node_slow_log_time_idx:0.000000;system_auth.role_members:0.000000;system_distributed.cdc_streams_descriptions_v2:0.000000;ks_truncate_large_partition.test_table:0.000000;system_traces.events:0.000000;system_distributed_everywhere.cdc_generation_descriptions_v2:0.000000;system_distributed.service_levels:1.000000;system_distributed.view_build_status:0.000000;keyspace1.standard1:0.362382;mview.users_by_first_name:0.492440;system_traces.sessions:0.000000;system_auth.role_permissions:0.000000;ks_truncate_large_partition.test_counters:0.000000;
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X4:1
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   RPC_ADDRESS:10.4.10.195
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   HOST_ID:741fdc78-2a04-4cbf-831e-cf7827a1daa5
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   RELEASE_VERSION:3.0.8
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X6:14
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X7:12
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   RACK:1c
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   NET_VERSION:0
< t:2024-01-20 14:46:32,386 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   DC:eu-west
```

on node 3
```
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG > /10.4.10.195
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   generation:1705760821
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   heartbeat:3441
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   HOST_ID:741fdc78-2a04-4cbf-831e-cf7827a1daa5
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   DC:eu-west
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X1:AGGREGATE_STORAGE_OPTIONS,ALTERNATOR_TTL,CDC,CDC_GENERATIONS_V2,COLLECTION_INDEXING,COMPUTED_COLUMNS,CORRECT_COUNTER_ORDER,CORRECT_IDX_TOKEN_
IN_SECONDARY_INDEX,CORRECT_NON_COMPOUND_RANGE_TOMBSTONES,CORRECT_STATIC_COMPACT_IN_MC,COUNTERS,DIGEST_FOR_NULL_VALUES,DIGEST_INSENSITIVE_TO_EXPIRY,DIGEST_MULTIPARTITION_READ,EMPTY_REPLICA_MUTATION_PAGES,EMPTY_REPLICA_PAGES,GROUP0_SC
HEMA_VERSIONING,HINTED_HANDOFF_SEPARATE_CONNECTION,INDEXES,LARGE_COLLECTION_DETECTION,LARGE_PARTITIONS,LA_SSTABLE_FORMAT,LWT,MATERIALIZED_VIEWS,MC_SSTABLE_FORMAT,MD_SSTABLE_FORMAT,ME_SSTABLE_FORMAT,NONFROZEN_UDTS,PARALLELIZED_AGGREG
ATION,PER_TABLE_CACHING,PER_TABLE_PARTITIONERS,RANGE_SCAN_DATA_VARIANT,RANGE_TOMBSTONES,ROLES,ROW_LEVEL_REPAIR,SCHEMA_COMMITLOG,SCHEMA_TABLES_V3,SECONDARY_INDEXES_ON_STATIC_COLUMNS,SEPARATE_PAGE_SIZE_AND_SAFETY_LIMIT,STREAM_WITH_RPC
_STREAM,SUPPORTS_RAFT_CLUSTER_MANAGEMENT,TABLE_DIGEST_INSENSITIVE_TO_EXPIRY,TOMBSTONE_GC_OPTIONS,TRUNCATION_TABLE,TYPED_ERRORS_IN_READ_RPC,UDA,UDA_NATIVE_PARALLELIZED_AGGREGATION,UNBOUNDED_RANGE_TOMBSTONES,UUID_SSTABLE_IDENTIFIERS,V
IEW_VIRTUAL_COLUMNS,WRITE_FAILURE_REPLY,XXHASH
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   NET_VERSION:0
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X8:v2;1705760937261;3fb3200f-e058-43dd-b5e0-4fc3f01ca205
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X5:1516:877238681:1705762027414
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X3:3
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   RELEASE_VERSION:3.0.8
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X9:org.apache.cassandra.locator.Ec2Snitch
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   LOAD:13030663891
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X4:1
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   RPC_ADDRESS:10.4.10.195
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X7:12
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X6:14
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   X2:system_auth.roles:1.000000;system_traces.node_slow_log:0.000000;system_distributed.cdc_generation_timestamps:0.000000;mview.users:0.994557;sy
stem_traces.sessions_time_idx:0.000000;system_auth.role_attributes:0.000000;mview.users_by_last_name:0.605232;system_traces.node_slow_log_time_idx:0.000000;system_auth.role_members:0.000000;system_distributed.cdc_streams_description
s_v2:0.000000;ks_truncate_large_partition.test_table:0.000000;system_traces.events:0.000000;system_distributed_everywhere.cdc_generation_descriptions_v2:0.000000;system_distributed.service_levels:1.000000;system_distributed.view_bui
ld_status:0.000000;keyspace1.standard1:0.373144;mview.users_by_first_name:0.532701;system_traces.sessions:0.000000;system_auth.role_permissions:0.000000;ks_truncate_large_partition.test_counters:0.000000;
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   SCHEMA:a4e0dc54-b797-11ee-f6f5-a538f551ea6e
< t:2024-01-20 14:47:09,026 f:cluster.py      l:2654 c:sdcm.cluster         p:DEBUG >   RACK:1c
```


Looking at generation and heartbeat, the output from node 3 is newer (because sct retrieved it at later time)
but node 3 is missing STATUS
This is exactly the same symptoms as in https://github.com/scylladb/scylladb/issues/16668

**Comment by kbr-scylla** [comment]
BTW
cdd5605d812929fb41f900ce5768f2db39aa80eb broke the output of gossipinfo

Now the numbers X<i> are assigned pretty much randomly

e.g. X7 in one node's gossipinfo output means a different state than X7 in another
Making the whole thing hard to debug

**Comment by kbr-scylla** [comment]
> https://github.com/scylladb/scylladb/commit/cdd5605d812929fb41f900ce5768f2db39aa80eb broke the output of gossipinfo

Or maybe not
I'm not sure I understand how these numbers are assigned
could be just that the lines are permuted, but the numbers stay the same

**Comment by kbr-scylla** [comment]
>  could be just that the lines are permuted, but the numbers stay the same

Confirmed

**Comment by aleksbykov** [comment]
Issue again reproduced with: 
## Packages
Scylla version: `5.5.0~dev-20240209.7a710425f0e2` with build-id `f4777533fdd4b53d42e7a22dd9605eb4feb4fc5a`
Kernel Version: `5.15.0-1053-aws`


## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

node5 seen new added node12 with same status UJ and with hostid ?
```
< t:2024-02-10 13:55:03,092 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.11.182  23.38 GB   256          ?       b9365636-1b5f-4687-8f91-eaa4bbd78c54  1c
< t:2024-02-10 13:55:03,119 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UJ  10.4.10.50   14.82 GB   256          ?       null                                  1c
< t:2024-02-10 13:55:03,128 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.10.99   19.35 GB   256          ?       52379a32-571c-4d02-8040-9604979c1441  1c
< t:2024-02-10 13:55:03,152 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.11.67   20.23 GB   256          ?       d26ffd30-0ba1-4ea9-aacd-deecc0e3364f  1c
< t:2024-02-10 13:55:03,152 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.9.79    21.77 GB   256          ?       4c8baa34-cddc-4d15-9857-1a1f358dd7f4  1c
< t:2024-02-10 13:55:03,152 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.8.120   13.28 GB   256          ?       f5742036-c8cf-4ba2-a2ed-ab7c198bf942  1c
< t:2024-02-10 13:55:03,152 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.11.249  20.71 GB   256          ?       d553f8a3-cca3-4477-ad0b-9a7ea9179d98  1c
< t:2024-02-10 13:55:03,152 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.9.57    13.87 GB   256          ?       7807eaac-e98f-4b93-92da-69328db65090  1c
< t:2024-02-10 13:55:03,152 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.11.217  20.02 GB   256          ?       bb007e77-e21c-4e7d-b83d-fb1815625a46  1c
< t:2024-02-10 13:55:03,152 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > 
< t:2024-02-10 13:55:03,152 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Note: Non-system keyspaces don't have the same replication settings, effective ownership information is meaningless
< t:2024-02-10 13:55:03,152 f:base.py         l:142  c:RemoteLibSSH2CmdRunner p:DEBUG > Command "/usr/bin/nodetool -u cassandra -pw 'cassandra'  status " finished with status 0
< t:2024-02-10 13:55:03,152 f:cluster.py      l:2539 c:sdcm.cluster_aws     p:DEBUG > Node longevity-tls-50gb-3d-master-db-node-ae910ac5-5 [54.217.138.87 | 10.4.11.182] (seed: True): Command '/usr/bin/nodetool -u cassandra -pw 'cassandra'  status ' duration -> 4.137540684998385 s
```

while node4 sees it UN.

In logs for node5 and node 12 i found that:

node5 start bootstrap for node 12 and marked it as finished:
```
2024-02-10T13:21:26.227+00:00 longevity-tls-50gb-3d-master-db-node-ae910ac5-5     !INFO | scylla[22724]:  [shard  0:strm] storage_service - bootstrap[ede025a8-3be8-4857-bb11-01aff07910c7]: Added node=10.4.10.50/e0387ef1-81f5-47d2-a39a-0e05c501fd16 as bootstrap, coordinator=10.4.10.50/e0387ef1-81f5-47d2-a39a-0e05c501fd16


2024-02-10T13:23:32.980+00:00 longevity-tls-50gb-3d-master-db-node-ae910ac5-5     !INFO | scylla[22724]:  [shard  0:strm] storage_service - bootstrap[ede025a8-3be8-4857-bb11-01aff07910c7]: Marked ops done from coordinator=10.4.10.50
```

while node12 started bootstrap from node5 but don't finish with with:
```
2024-02-10T13:21:26.104+00:00 longevity-tls-50gb-3d-master-db-node-ae910ac5-12     !INFO | scylla[6196]:  [shard  0:strm] storage_service - bootstrap[ede025a8-3be8-4857-bb11-01aff07910c7]: Added node=10.4.10.50/e0387ef1-81f5-47d2-a39a-0e05c501fd16 as bootstrap, coordinator=10.4.10.50/e0387ef1-81f5-47d2-a39a-0e05c501fd16
2024-02-10T13:21:26.104+00:00 longevity-tls-50gb-3d-master-db-node-ae910ac5-12     !INFO | scylla[6196]:  [shard  0:strm] node_ops - bootstrap[ede025a8-3be8-4857-bb11-01aff07910c7]: Finished bootstrap_prepare[ede025a8-3be8-4857-bb11-01aff07910c7]: ignore_nodes={}, leaving_nodes={}, replace_nodes={}, bootstrap_nodes={{10.4.10.50 -> {-9068049917611146609, 3220989120584746042, -5223742092451205683, -4621444056473814661, -1467756090397522078, 8742611262885005949, -954124957226714107, -3437741198965068571, -3802428319303067030, -3406177346827464984, 8842752507575225947, 2918081643325006772, 8550566883745357906, 7135434246118972521, 6349767850945099357, 3931984704487769572, 423128864807101177, -4132980784646018006, -1950084788808370605, 1090323881290516490, 5806355568040337072, 7067803903413696787, 6111597148510090125, -5117470981665064146, -25072091412691687, 87995873340134978, -3253399957638296615, -6596469020037302159, 4277578574392022563, -313187710106285707, -6102005041380846004, 3134046286976944005, 8940919647441404882, -3363881565187881498, 7035762436437810123, 1936217798557655892, 3156281457276156872, -3379117964937816366, -4080337871225641779, 6404295616923189514, -673863018275268733, 5729740366909214536, -5351692409291749758, -3258920311565612270, 5694414160550233708, -1397600111064313138, 6703741213929951203, 6505862930764312722, -3276128582739133887, -8447653426424905539, 128665423754075677, 6724839106644196200, 2154147920524314487, 323614088949063951, 3847657996734488802, -3633758726675593041, 5078547278252267231, -2473578751605277598, 3571561970289582041, 4565356945319459751, -8456607964291856434, -356747815457653349, 7460141472928558931, 20475686670595879, 1323633295539768926, -8519302990175776362, 8602065883804349687, 8502677393857968385, -8220813578483265606, -7975366545360412225, 1145294996589644850, 1040314123403253820, 552585384777952157, -2596268033788224386, -1598785830545333540, 7611476126215791551, -1266425945068519261, 7367352256169634073, 7715483634084307648, 1255344574293079293, 4897898281699583425, -6339663558515279765, 9195976485244466660, 2382031451538695997, -3827692919099001444, 3036955341192847808, -3756421446425436672, -922458656432482896, 7177095155757100646, -3383552684371074701, -5365362512406201422, 7625657948623220385, 8836567972560272387, -524557003438580658, -4656829924690380735, 7336273056229298044, 1350724786856190800, -4418804642396046850, -7629008349979896580, -6984151970367519674, 4918336857414699756, 5620861608600340176, -9013341237604893920, 2885216463895313386, -4888027151983964661, 3310690120022031672, -350938665669384642, 5086830017847456624, 7988241684609890375, 3274442674623671580, -7950844134606087747, 655658345546781749, -8794281667839919093, -1156273156345813185, 3199513536228312864, -4810357811600411697, -4554467893946550404, -709871614378245259, 6910398451030485587, 7462853830425862775, 5315543532016967616, 8623502531830069522, -708797415324083358, -8364226345266445041, 9013388181965065397, -3804685501194338000, 3438345535657851622, -2310054378935953800, -3450676107446885720, 4192450881182229170, -6160409032791833727, 5556763107417799857, -1252250756289989744, 7036540918237075759, -2117994774018077181, -2670457008234917151, 7520650633263734760, -1862070515684505578, 8607639165027846786, 1374900775722423230, 8600839489245583223, 8359181379219711599, -2598714773396998019, -40434029744124835, -8555569256268106546, -3548317464039620788, 7933046226380921658, 7759795783356864043, -950939101649478781, -3856097669220029985, 5767152223560782959, -7234536689004728520, -4764917629732337597, -8718561836957718983, -7721386384827051916, 366241612163642777, -3704898244036322775, -3702710561702212244, -5148428119827152108, -2740737198994327611, -7580466022102261900, -1318118997563461834, 927313104829565784, -4274805081851537012, 5461410730755543330, 2707232751630020470, 7236323115174448998, 9149822446308259053, 2992505309506206095, 755121478313908425, 3348836275798528903, 4339823570080978199, 5153328200461343987, 2136068122799400103, -5643145190002012931, -6778551110214887744, 3686043604525679487, -1660927634542521620, -2485955868319398915, -9027553773563801560, 1469492999501780476, -1296648245083047906, -1961608052899827345, 987086462789495705, -8894858601717061443, -4839921404505940328, -6089703963362764511, 7832899009491941867, -6486780076379974524, -7765759770075008800, 2781079163246918422, -2659557543632580104, -3121581127038981697, -5270096600554700840, 2638194481857651664, -6442916404659554902, -8765300730462314322, -2183481516131057921, -1748573274181109220, -1113268055306570187, -4193798862921391497, 1155884411519062310, -1450671925422679784, -5274920032602472152, -6407875865253367105, 4274305547476399061, 584408418745245120, 601495832888272803, 6509005252212348026, -6563511139455554122, 1518637527308978575, 7971551302498147397, -6656683596997820907, 3297722060575721562, 4648949465617835270, -2990891102135347564, 581744491678157332, 8939357972226879728, 459018843722522636, -3606331752838386701, 557733246161676344, -8781706920954092286, -2814924710993195077, 4428929209377626409, 4584895258779621722, -3715371545763730172, 7769627010366279911, -852197851159661667, -4918800416862249051, 2397044144018924596, -4252478251937972393, 675082420572845932, -4817684013691259059, -9162233293811743550, -6942312461706700545, -3556796637648763350, -6541757679277988203, 2746937689937932664, -137085740818033971, -3778060135588775653, 2728230328155587672, -7844653013916819554, -133614690738122080, -5102542999487687779, 972902802121679103, -5216201473177611730, 7435258821120710544, -7240956111176763473, -3172357634930535264, 5431446740685520143, -5846204079638282943, 626041271289954800, -1870293561209602930, 1414202598130267370, -1755904192182968176, 3812860850734643119}}}, repair_tables={}

```
and there is no info about finished bootstrap repair



## Impact

*Describe the impact this issue causes to the user.*

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details

Cluster size: 6 nodes (i4i.4xlarge)

Scylla Nodes used in this run:
- longevity-tls-50gb-3d-master-db-node-ae910ac5-1 (54.74.194.77 | 10.4.11.25) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-10 (34.242.89.36 | 10.4.10.99) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-11 (3.249.135.84 | 10.4.9.57) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-12 (63.32.106.146 | 10.4.10.50) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-13 (3.253.96.134 | 10.4.8.120) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-2 (34.247.90.166 | 10.4.9.24) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-3 (3.255.78.13 | 10.4.10.133) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-4 (54.229.208.191 | 10.4.9.79) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-5 (54.217.138.87 | 10.4.11.182) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-6 (54.170.18.177 | 10.4.11.249) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-7 (34.249.207.143 | 10.4.11.217) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-8 (54.78.67.156 | 10.4.11.113) (shards: 14)
- longevity-tls-50gb-3d-master-db-node-ae910ac5-9 (18.201.47.84 | 10.4.11.67) (shards: 14)


OS / Image: `ami-07569b2b3deaa1ea9` (aws: undefined_region)

Test: `longevity-50gb-3days-test`
Test id: `ae910ac5-10ac-43d8-a6ed-4c528ddd4715`
Test name: `scylla-master/longevity/longevity-50gb-3days-test`
Test config file(s):

- [longevity-50GB-3days-authorization-and-tls-ssl.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/fba16698a8f23c0c1457cc8e9c268d371f4b3d6c/test-cases/longevity/longevity-50GB-3days-authorization-and-tls-ssl.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor ae910ac5-10ac-43d8-a6ed-4c528ddd4715`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=ae910ac5-10ac-43d8-a6ed-4c528ddd4715)
- Show all stored logs command: `$ hydra investigate show-logs ae910ac5-10ac-43d8-a6ed-4c528ddd4715`


## Logs:
- **db-cluster-ae910ac5.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/db-cluster-ae910ac5.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/db-cluster-ae910ac5.tar.gz)
- **sct-runner-events-ae910ac5.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/sct-runner-events-ae910ac5.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/sct-runner-events-ae910ac5.tar.gz)
- **sct-ae910ac5.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/sct-ae910ac5.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/sct-ae910ac5.log.tar.gz)
- **loader-set-ae910ac5.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/loader-set-ae910ac5.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/loader-set-ae910ac5.tar.gz)
- **monitor-set-ae910ac5.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/monitor-set-ae910ac5.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/ae910ac5-10ac-43d8-a6ed-4c528ddd4715/20240210_140747/monitor-set-ae910ac5.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-50gb-3days-test/235/)
[Argus](https://argus.scylladb.com/test/5df88bc6-3ede-4a28-9586-773f93374239/runs?additionalRuns[]=ae910ac5-10ac-43d8-a6ed-4c528ddd4715)
</details>

**Comment by mykaul** [comment]
@kbr-scylla - what's the next step here?

**Comment by kbr-scylla** [comment]
You reminded me that I need to backport https://github.com/scylladb/scylladb/pull/17499 to 5.4 and 5.2

**Comment by mykaul** [comment]
How is https://github.com/scylladb/scylladb/issues/17857 related to this?

**Comment by kbr-scylla** [comment]
https://github.com/scylladb/scylladb/issues/17857 is just https://github.com/scylladb/scylladb/issues/16902#issuecomment-1924074075 as a separate issue. (I should have created it earlier.)

It is a minor problem detected in this report, unrelated to the larger problem (of node remaining in UJ.)
